services/debugger/source/eventUploader.go

```diff
@@ -107,13 +102,15 @@ func RecordEvent(writeKey, eventBatch string) bool {
 }
 
 func (*EventUploader) Transform(data interface{}) ([]byte, error) {
-	eventBuffer := data.([]interface{})
+	eventBuffer, ok := data.([]GatewayEventBatchT)
```

Comment:
why are we accepting `any` in the method signature whereas we are very specific to what we expect?!

Reply:
`Uploader` module which does the final upload to config backend has access to `Transform` interface and the data type which uploader receives for source/destination/transformation live events are different. So interface has a generic  signature but the actual implementation checks for the exact data struct it needs

Reply:
This doesn't sound convincing, imo it is an indication of a weak design and poorly chosen exported API & responsibilities for these two components.

Reply:
Anything which you have in mind for this?